config/samples/oas/petstore.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: v1
+data:
+  test_oas.json: |-
+    {
+        "info": {
+          "title": "Petstore",
+          "version": "1.0.0"
+        },
+        "openapi": "3.0.3",
+        "components": {},
+        "paths": {},
+        "x-tyk-api-gateway": {
```

Comment:
Make sense

Reply:
I was thinking of building my own operator for this functionality. but I'm glad to see that this is becoming default functionality. The only bit I'm missing/would change that would solve our companies use case is instead of the config map having the OAS definition itself it would have a URL to where to download the definition form. that way when a team create a micro service the want to automatically expose via tyk the don't need a additional manual step of coping their definition in a config map. or having ci/cd do that for them.

A change of the annotation on the TykOAS will be enough to trigger a reload of this all. 

Their might also be a nice gain if the TykOAS would also include an ApiDefinition with the same [template ](template: "true") flags as used for an ingress to make a merge between missing features that can't be defined in OAS but are TYK features.

Sorry if this is not the place for this feature req. if you want it moved i can do so.

Reply:
Thank you @g-bohncke for such early feedback on the implementation. It always great to use-case of users.
Our plan was to support configMap in MVP and extend in future to support different sources.
It would be great if you raise a Feature Request on Github so we can discuss about implementation there.

> Their might also be a nice gain if the TykOAS would also include an ApiDefinition with the same template flags as used for an ingress to make a merge between missing features that can't be defined in OAS but are TYK features.
In our upcoming 5.3 release, OAS will support majority of Classic API features.